src/index.ts

```diff
@@ -45,7 +44,7 @@ class Webhooks<TTransformed> {
       throw new Error("[@octokit/webhooks] options.secret required");
     }
 
-    const state: State = {
+    const state = {
```

Comment:
I think this should remain, since it should be conforming to the `State` type (which itself is in a bit of a mess)

Reply:
I'm in favor of dropping it because the inferred type is narrower than `State`:
- It's assignable to `State` (the following parameters are `State`) https://github.com/octokit/webhooks.js/blob/dd65cf5e97225a8a216af818a4d7f8c830621674/src/index.ts#L62-L63
- The explicit type discards information: the fact that `eventHandler` is non-null

That said it's a minor detail, give the word and I'll make this change.

Reply:
I hear what you're saying, and am not wholly against it - however, I always tend to prefer explicitly typing object and arrays in most cases, as it makes it clearer what's in the wrong when there's an error (since it'll error on this assignment being wrong, rather than the usage being wrong), which actually leads me to my second point:

> The explicit type discards information: the fact that eventHandler is non-null

Exactly, which may or may not be correct and is effectively the tl;dr of #425 😅 

`State` looks to me like the classic genetically named internal-only state type that tend to popup when you move from JS to TS.

Because of that, I'd prefer to keep the explicit type annotation, to capture what type we're aiming for and hopefully make it easier to spot where our inconsistencies are when refactoring that type.

Reply:
> `State` looks to me like the classic genetically named internal-only state type that tend to popup when you move from JS to TS.

That is pretty much what happened there when I initially moved to TS.

TBF, we haven't really done much refactoring since, I think it's time to revisit a lot of this stuff with the new experiences we have and new TS features as well

Reply:
Okay, thanks for this thoughtful explanation. Done :heavy_check_mark: